From 5c8b50014acd09f112323fb3f7aa5f2f1ec5fa61 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Fri, 25 May 2012 13:49:48 +0300
Subject: [PATCH] Fix check of Handles

A Handle should be checked using IsNull() method and not directly
by its name as if it was a NULL pointer.
---
 src/NIS/NIS_Surface.cxx                      |    2 +-
 src/ViewerTest/ViewerTest_ViewerCommands.cxx |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: oce/src/NIS/NIS_Surface.cxx
===================================================================
--- oce.orig/src/NIS/NIS_Surface.cxx
+++ oce/src/NIS/NIS_Surface.cxx
@@ -486,7 +486,7 @@ void  NIS_Surface::SetDisplayMode (const
       isUpdate = Standard_True;
     }
   }
-  if (isUpdate && GetDrawer()) {
+  if (isUpdate && !GetDrawer().IsNull()) {
     const Handle(NIS_SurfaceDrawer) aDrawer =
       static_cast<NIS_SurfaceDrawer *>(DefaultDrawer(0L));
     aDrawer->Assign (GetDrawer());
Index: oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
===================================================================
--- oce.orig/src/ViewerTest/ViewerTest_ViewerCommands.cxx
+++ oce/src/ViewerTest/ViewerTest_ViewerCommands.cxx
@@ -2365,7 +2365,7 @@ V3d_LineItem::V3d_LineItem(Standard_Real
   myX1(X1), myY1(Y1), myX2(X2), myY2(Y2), myLayerMgr(theLayerMgr),
   myType(theType), myWidth(theWidth), myTransparency(theTransp)
 {
-  if (myLayerMgr && myLayerMgr->Overlay())
+  if (myLayerMgr && !myLayerMgr->Overlay().IsNull())
     myLayerMgr->Overlay()->AddLayerItem (this);
 }
 
